### PR TITLE
UploadAsync updated comment, code and UT

### PR DESCRIPTION
**Pull request by marek-vysoky**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

**Comment by coolboy** [comment]
@marek-vysoky conflict here

**Comment by marek-vysoky** [comment]
I've merged master to this branch, resolved the conflict and pushed the changes

File: Box.V2.JWTAuth/BoxJWTAuth.cs [hunk 1]
```diff
@@ -17,6 +17,12 @@
 
 namespace Box.V2.JWTAuth
 {
+    ///<summary>
+    /// Box’s new authentication model allows applications to authenticate directly to Box using a JSON Web Token (JWT) signed with an RSA key. This authentication method is meant for server-to-server applications and replaces the first leg of the standard 3-legged OAuth 2.0 process in which users grant an application authorization to access the their Box account
+    ///</summary>
+    ///<remarks>
+    ///https://docs.box.com/docs/getting-started-box-platform
```
Comment (coolboy) [inline Box.V2.JWTAuth/BoxJWTAuth.cs]: So for the comments, the style is to have a space after ///

File: Box.V2/Models/BoxPermission.cs [hunk 2]
```diff
@@ -17,7 +17,6 @@ public class BoxPermission
         /// <summary>
         /// Whether the item can be previewed or not
         /// </summary>
-        [Obsolete("CanPreview is now deprecated in the API and will always return true.")]
```
Comment (coolboy) [inline Box.V2/Models/BoxPermission.cs]: why removing this?

Comment (valer-kovac) [inline Box.V2/Models/BoxPermission.cs]: I've checked it against specification. There is no mark for obsolete for this item.

**Comment by cburnette** [comment]
@coolboy @marek-vysoky Conflicts now due to recent merges.

**Comment by marek-vysoky** [comment]
@cburnette Conflict resolved

**Pull request merged**